ops/testing.py

```diff
@@ -815,6 +815,19 @@ def set_leader(self, is_leader: bool = True) -> None:
         if is_leader and not was_leader and self._charm is not None and self._hooks_enabled:
             self._charm.on.leader_elected.emit()
 
+    def set_planned_units(self, num_units: int) -> None:
+        """Set the number of "planned" units  that "Application.planned_units" should return.
+
+        In real world circumstances, this number will be the number of units in the
+        application. E.g., this number will be the number of peers this unit has, plus one, as we
+        count our own unit in the total.
+
+        At the time of this writing (2021/11), a change to the return from planned_units will not
+        generate an event. Typically, a charm author would check planned units during a config or
+        install hook, or after receiving a peer relation joined event.
+        """
+        self._backend._planned_units = num_units
```

Comment:
Do we want a quick check for negative/garbage values here and raise a `TypeError`?

Reply:
We should check for negative values. 0 is valid, as are all the positive integers (I think), and everything else should be caught by the type checker.

I'll add a check for negative vals ...